### PR TITLE
chore: add error type for non retryable errors

### PR DESCRIPTION
**Pull request by stoksc**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/sproto/resources.go [hunk 1]
```diff
@@ -224,12 +224,31 @@ func FromContainerFailureType(t aproto.FailureType) FailureType {
 	}
 }
 
-// IsRestartableSystemError checks if the error is caused by the system and
+// InvalidResourcesRequestError is an unrecoverable validation error from the underlying RM.
+type InvalidResourcesRequestError struct {
```
Comment (stoksc) [inline master/internal/sproto/resources.go:228]: To be used by slurm

File: master/internal/sproto/resources.go [hunk 2]
```diff
@@ -224,12 +224,31 @@ func FromContainerFailureType(t aproto.FailureType) FailureType {
 	}
 }
 
-// IsRestartableSystemError checks if the error is caused by the system and
+// InvalidResourcesRequestError is an unrecoverable validation error from the underlying RM.
```
Comment (ioga) [inline master/internal/sproto/resources.go:227]: validation of what?

File: master/internal/sproto/resources.go [hunk 3]
```diff
@@ -224,12 +224,31 @@ func FromContainerFailureType(t aproto.FailureType) FailureType {
...
+type InvalidResourcesRequestError struct {
+	Cause error
+}
+
+func (e InvalidResourcesRequestError) Error() string {
+	return fmt.Sprintf("unrecoverable system error: %s", e.Cause.Error())
```
Comment (ioga) [inline master/internal/sproto/resources.go:233]:
```suggestion
	return fmt.Sprintf("invalid resources request: %s", e.Cause.Error())
```

**Review by ioga — APPROVED** [review]

File: master/internal/sproto/resources.go (hunk 2 above)
Comment (stoksc) [reply master/internal/sproto/resources.go:227]: if someone tries to submit a job in a way that slurm rejects, we can catch it and prevent 5 retries https://github.com/determined-ai/determined-ee/pull/150/files#diff-8fffae72479c7eb5e7d37769d8b13e5867fae19ad9b63128bb727107c6b335d6R32

Comment (stoksc) [reply master/internal/sproto/resources.go:227]: trying to accomplish ^ with typed errors instead of regex'ing. also helps prevent rebase hell.

**Pull request merged**